src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Behaviors/ListSearchPipeBehavior.cs

```diff
@@ -0,0 +1,131 @@
+using Microsoft.Health.Fhir.Core.Features;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Features.Search.SearchValues;
+using Microsoft.Health.Fhir.Core.Messages.Search;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Features.Search.Behavior
+{
+    /// <summary>
+    /// This behavior looks for searches that use _list then resolves the list and rewrites the query with the resource _ids
+    /// </summary>
+    public class ListSearchPipeBehavior : IPipelineBehavior<SearchResourceRequest, SearchResourceResponse>
+    {
+        private readonly IScoped<IFhirDataStore> _dataStore;
+        private readonly ResourceDeserializer _deserializer;
+        private readonly IBundleFactory _bundleFactory;
+        private readonly IReferenceSearchValueParser _referenceSearchValueParser;
+        private readonly ISearchOptionsFactory _searchOptionsFactory;
+        private Lazy<SearchResourceResponse> _emptyResponse;
+
+        public ListSearchPipeBehavior(
+            ISearchOptionsFactory searchOptionsFactory,
+            IBundleFactory bundleFactory,
+            IScoped<IFhirDataStore> dataStore,
+            ResourceDeserializer deserializer,
+            IReferenceSearchValueParser referenceSearchValueParser)
+        {
+            EnsureArg.IsNotNull(bundleFactory, nameof(bundleFactory));
+            EnsureArg.IsNotNull(dataStore, nameof(dataStore));
+            EnsureArg.IsNotNull(deserializer, nameof(deserializer));
+            EnsureArg.IsNotNull(referenceSearchValueParser, nameof(referenceSearchValueParser));
+            EnsureArg.IsNotNull(searchOptionsFactory, nameof(searchOptionsFactory));
+
+            _searchOptionsFactory = searchOptionsFactory;
+            _bundleFactory = bundleFactory;
+            _dataStore = dataStore;
+            _deserializer = deserializer;
+            _referenceSearchValueParser = referenceSearchValueParser;
+        }
+
+        public async Task<SearchResourceResponse> Handle(SearchResourceRequest request, CancellationToken cancellationToken, RequestHandlerDelegate<SearchResourceResponse> next)
+        {
+            EnsureArg.IsNotNull(request, nameof(request));
+
+            Tuple<string, string> listParameter = request.Queries
+                .FirstOrDefault(x => string.Equals(x.Item1, KnownQueryParameterNames.List, StringComparison.OrdinalIgnoreCase));
+
+            // if _list was not requested, continue...
+            if (listParameter == null)
+            {
+                return await next();
+            }
+
+            // Remove the 'list' params from the queries, to be later replaced with specific ids of resources
+            IEnumerable<Tuple<string, string>> query = request.Queries.Except(new[] { listParameter });
+
+            // prepare an empty response (lazy) in case we need it
+            _emptyResponse = new Lazy<SearchResourceResponse>(() => CreateEmptySearchResponse(request));
```

Comment:
I don't think `_emptyResponse` should be a field unless can utilize a field or constructor initializer. To keep things simple its probably ok to just `return CreateEmptySearchResponse(request)` when needed